src/createBaseRouter.js

```diff
@@ -41,7 +41,7 @@ export default function createBaseRouter({ routeConfig }) {
     }) {
       const currentMatchIndex = this.matchIndex;
       const routes = getRoutes(routeConfig, match);
-      const fullMatch = { ...match, routes, context: matchContext };
+      const fullMatch = { ...match, routes, matcher, context: matchContext };
```

Comment:
`getComponent`, `getData`, and `render` now have access to `matcher`. Should be okay, right?